kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -171,6 +173,21 @@ def foundry_kompile(
             ccopts=ccopts,
             llvm_kompile=llvm_kompile,
         )
+        if llvm_library:
+            _LOGGER.info(f'Kompiling definition to LLVM dy.lib: {foundry_main_file}')
+            KEVM.kompile(
+                foundry_llvm_dir,
+                KompileBackend.LLVM,
+                foundry_main_file,
+                emit_json=True,
+                includes=includes,
+                main_module_name=main_module,
+                syntax_module_name=syntax_module,
+                md_selector=md_selector,
+                debug=debug,
+                ccopts=ccopts,
+                llvm_kompile_type=LLVMKompileType.C,
+            )
```

Comment:
Consider extracting a nested function that captures arguments shared between the two calls, and exposes the rest as parameters.